tensorly/decomposition/candecomp_parafac.py

```diff
@@ -42,14 +42,19 @@ def initialize_factors(tensor, rank, init='svd', svd='numpy_svd', random_state=N
     """
     rng = check_random_state(random_state)
     eps = tl.eps(tensor.dtype)
+    weights = tl.ones(rank, **tl.context(tensor))
 
     if init == 'random':
         factors = [tl.tensor(rng.random_sample((tensor.shape[i], rank)), **tl.context(tensor)) for i in range(tl.ndim(tensor))]
         if non_negative:
             factors = [tl.abs(f) for f in factors]
-        if normalize_factors: 
-            factors = [f/(tl.reshape(tl.norm(f, axis=0), (1, -1)) + eps) for f in factors]
-        return factors
+        if normalize_factors:
```

Comment:
Here we could just used `kruskal_normalize`: 
https://github.com/tensorly/tensorly/blob/cb099d9045c2ed5b40dc03d2d7c4887d7079c05a/tensorly/kruskal_tensor.py#L104

I remember having some issues with the normalization in some edge cases -- is there any strong case for just adding eps vs actually checking for zero as in `kruskal_normalize`?

Reply:
Maybe `eps` can just be removed? It seems like the issue would be if the factor norm is 0.0, but if that's the case I'm not sure what to do besides throw an error... I could add an assert to check for 0.0 so that the error is clearer?

Reply:
I meant that we could simply return tensorly.kruskal_tensor.normalize_kruskal((weights, factors)), this would normalize (by checking for zero values), but, I am not 100% sure whether that is better than just adding eps.
In any case, it seems more consistent to always use that to normalize, so if we change it there, it's updated for all functions.